src/linear_algebra/affine_space.lean

```diff
@@ -289,4 +346,134 @@ def comp (f : affine_map k V2 P2 V3 P3) (g : affine_map k V1 P1 V2 P2) :
   f.comp g p = f (g p) := rfl
 
+@[simp] lemma comp_id (f : affine_map k V1 P1 V2 P2) : f.comp (id k V1 P1) = f := ext $ λ p, rfl
+
+@[simp] lemma id_comp (f : affine_map k V1 P1 V2 P2) : (id k V2 P2).comp f = f := ext $ λ p, rfl
+
+lemma comp_assoc (f₃₄ : affine_map k V3 P3 V4 P4) (f₂₃ : affine_map k V2 P2 V3 P3)
+  (f₁₂ : affine_map k V1 P1 V2 P2) :
+  (f₃₄.comp f₂₃).comp f₁₂ = f₃₄.comp (f₂₃.comp f₁₂) :=
+rfl
+
+instance : monoid (affine_map k V1 P1 V1 P1) :=
+{ one := id k V1 P1,
+  mul := comp,
+  one_mul := id_comp,
+  mul_one := comp_id,
+  mul_assoc := comp_assoc }
+
+@[simp] lemma coe_mul (f g : affine_map k V1 P1 V1 P1) : ⇑(f * g) = f ∘ g := rfl
+@[simp] lemma coe_one : ⇑(1 : affine_map k V1 P1 V1 P1) = _root_.id := rfl
+
+/-- The affine map from `k` to `P1` sending `0` to `p` and `1` to `v +ᵥ p`. -/
+def line_map (p : P1) (v : V1) : affine_map k k k V1 P1 :=
+{ to_fun := λ c, c • v +ᵥ p,
+  linear := linear_map.id.smul_right v,
+  map_vadd' := λ a b, by simp [add_smul, add_action.vadd_assoc] }
+
+lemma line_map_apply (p : P1) (v : V1) (c : k) : line_map p v c = c • v +ᵥ p := rfl
+
+@[simp] lemma line_map_linear (p : P1) (v : V1) :
+  (line_map p v : affine_map k k k V1 P1).linear = linear_map.id.smul_right v :=
+rfl
+
+@[simp] lemma line_map_zero (p : P1) : line_map p (0:V1) = const k k k V1 p :=
+by { ext c, simp [line_map_apply] }
+
+@[simp] lemma line_map_apply_zero (p : P1) (v : V1) : line_map p v (0:k) = p :=
+by simp [line_map_apply]
+
+@[simp] lemma affine_apply_line_map (f : affine_map k V1 P1 V2 P2) (p : P1) (v : V1) (c : k) :
+  f (line_map p v c) = line_map (f p) (f.linear v) c :=
+by simp [line_map_apply]
+
+@[simp] lemma affine_comp_line_map (f : affine_map k V1 P1 V2 P2) (p : P1) (v : V1) :
+  f.comp (line_map p v) = line_map (f p) (f.linear v) :=
+ext $ f.affine_apply_line_map p v
+
+lemma line_map_vadd_neg (p : P1) (v : V1) :
+  line_map (v +ᵥ p) (-v) = (line_map p v).comp (line_map (1:k) (-1:k)) :=
+by { rw [affine_comp_line_map], simp [line_map_apply] }
+
 end affine_map
+
+namespace affine_map
+
+variables {k : Type*} {V1 : Type*} {P1 : Type*} {V2 : Type*} [comm_ring k]
+    [add_comm_group V1] [module k V1] [affine_space k V1 P1] [add_comm_group V2] [module k V2]
+
+/-- If `k` is a commutative ring, then the set of affine maps with codomain in a `k`-module
+is a `k`-module. -/
+instance : module k (affine_map k V1 P1 V2 V2) :=
+{ smul := λ c f, ⟨c • f, c • f.linear, λ p v, by simp [smul_add]⟩,
+  one_smul := λ f, ext $ λ p, one_smul _ _,
+  mul_smul := λ c₁ c₂ f, ext $ λ p, mul_smul _ _ _,
+  smul_add := λ c f g, ext $ λ p, smul_add _ _ _,
+  smul_zero := λ c, ext $ λ p, smul_zero _,
+  add_smul := λ c₁ c₂ f, ext $ λ p, add_smul _ _ _,
+  zero_smul := λ f, ext $ λ p, zero_smul _ _ }
+
+@[simp] lemma coe_smul (c : k) (f : affine_map k V1 P1 V2 V2) : ⇑(c • f) = c • f := rfl
+
+variable (V1)
+
+/-- `dilation V c r` is the dilation about `c` with scale factor `r`. -/
```

Comment:
Is `dilation` or `homothety` more standard? I would have picked `dilation` myself, but it seems many people use `dilation` for the group generated by translations and homotheties. (So in particular translations are dilations, which doesn't fit the definition here.)

Reply:
I copied the name from https://github.com/leanprover-community/sphere-eversion/pull/8

Reply:
This name was from Scott. I would definitely write homethety because that sounds like a translation from the French homothétie. If dilation is ambiguous and homothety is an actual English word then I'd go with homothety.

Reply:
Done.